packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -1487,8 +1486,7 @@ class _CupertinoDatePickerDateState extends State<CupertinoDatePicker> {
 
     final double maxPickerWidth = totalColumnWidths > _kPickerWidth ? totalColumnWidths : _kPickerWidth;
 
-    return MediaQuery(
-      data: MediaQuery.of(context).copyWith(textScaleFactor: 1.0),
+    return MediaQuery.disableTextScaling(
```

Comment:
is the disableTextScaling method necessary? It looks like either way of setting textScaleFactor = 1.0 or textScaler.noScaling would work. I assume `disableTextScaling` is here so that there will just be one way to disable scaling, but once `textScaleFactor` is removed, this method will be redundant.

Reply:
That's a convenience method so yes it doesn't do anything special. But I think the implementation is slightly better than just getting `MediaQuery.of` from CupertinoDatePicker's context, since when `MediaQueryData` changes only the `Builder` widget will rebuild (instead of the date picker widget).